src/python/pants/notes/master.rst

```diff
@@ -4,6 +4,63 @@ Master Pre-Releases
 This document describes development releases which occur weekly from master, and which have
 not undergone the vetting associated with ``stable`` releases.
 
+1.21.0.dev0 (09/06/2019)
+------------------------
+
+API Changes
+~~~~~~~~~~~
+
+* New intrinsic InputFileContent -> Digest (#8226)
+  `PR #8226 <https://github.com/pantsbuild/pants/pull/8226>`_
+
+Bugfixes
+~~~~~~~~
+
+* fix exception handling of KeyboardInterrupt during startup (#8182)
+  `PR #8182 <https://github.com/pantsbuild/pants/pull/8182>`_
+
+* Use correct function on list (#8238)
+  `PR #8238 <https://github.com/pantsbuild/pants/pull/8238>`_
+
+* Support omitting the test sources themselves from coverage. (#8208)
+  `PR #8208 <https://github.com/pantsbuild/pants/pull/8208>`_
+
+* Use NGContext.exit() instead of sys.exit(). (#8239)
+  `PR #8239 <https://github.com/pantsbuild/pants/pull/8239>`_
+
+* Disable pantsd in inner runs. (#7884)
+  `PR #7884 <https://github.com/pantsbuild/pants/pull/7884>`_
+
+* Fix error when uploading stats to stats storage (#8233)
+  `PR #8233 <https://github.com/pantsbuild/pants/pull/8233>`_
+
+* Fix custom panic handler (#4884) (#8219)
+  `PR #4884 <https://github.com/pantsbuild/pants/pull/4884>`_
+  `PR #8219 <https://github.com/pantsbuild/pants/pull/8219>`_
+
+Refactoring, Improvements, and Tooling
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+* Quash warnings about missing dyn in generated code (#8245)
+  `PR #8245 <https://github.com/pantsbuild/pants/pull/8245>`_
+
+* Split store into lib, local, remote modules (#8244)
+  `PR #8244 <https://github.com/pantsbuild/pants/pull/8244>`_
+
+* Use `./pants.pex`, not `./pants`, to run internal integration tests (#8183)
```

Comment:
This should be API change because it impacts any users of Pants as a library. Twitter and Foursquare are impacted by this for example.

Reply:
How so? The integration test base class still falls back to using the ./pants file; so external users testing their loose source or dist plugins using the integration test base class should be unaffected.

Reply:
If they have pants.pex in their build root, they will end up using that. Not everyone may realize this and it’s plausible someone has an outdated pants.pex that causes strange failures due to this change. 

So the question becomes how likely is it that end users have pants.pex in their build root? I have no idea. If it’s very unlikely, I agree then that it’s okay to keep here.

Reply:
I think its very unlikely. This has been a good reading exercise for me though. I think the right way to do this so it works as a lib, dependency stuff included for chroot / v2 is to do something like this:
https://github.com/pantsbuild/pants/blob/874cd09a08453fd1acb745afb098db1c57420ca3/contrib/python/src/python/pants/contrib/python/checks/tasks/checkstyle/checkstyle.py#L131-L151

IOW, if using pants as a library, pants is coming from its wheel. The proper dependency is already expressed and the full pants binary contents are in your chroot by virtue of this. As such, the right way to invoke pants in that case is to find its entrypoint via pkg_resources and invoke that instead of a pants script / pex:
```
$ unzip -qc pantsbuild.pants-1.20.0rc0-cp36-abi3-manylinux1_x86_64.whl pantsbuild.pants-1.20.0rc0.dist-info/entry_points.txt
[console_scripts]
pants = pants.bin.pants_loader:main
```

I'll see if I can find time to try this out this weekend.